server/api/order.js

```diff
@@ -1,64 +1,144 @@
 const router = require('express').Router()
-const {Order, OrderProduct} = require('../db/models')
+const {Order, OrderProduct, Product} = require('../db/models')
 
+//GET all orders and their associated product and orderProduct info
+router.get('/', async (req, res, next) => {
+  try {
+    const orders = await Order.findAll({
+      include: {
+        model: Product,
+        through: OrderProduct 
+      }
+    })
+    res.json(orders)
+  } catch (err) { next (err) }
+
+})
+
+//POST new order for session user
 router.post('/', async (req, res, next) => {
   try {
-    const newOrder = await Order.create({ userId: req.user.id })
+    const order = await Order.create({ 
+        userId: req.user.id,
+        status:false
+      })
+    res.json(order)
+  } catch (err) {
+    next(err)
+  }
+})
+
```

Comment:
For all the routes with the orderId appended in the URL... 

Still fuzzy on how we know the orderId to view the cart... I know when we create an order, an id will be created, but how will the URL know what order it is when the user is making their order and not clicking on a specific order to view the cart?  Maybe we have a cart button that brings the user to a page with all their orders and then they click on the order that is in progress (so when you add your first item in the cart, the user order page automatically adds the order POST creates)?  

Or another thought would be that instead of findByPk, we use a Order.findAll where status is false in our routes for cart?

Reply:
I was fuzzy about it yesterday but did some thinking last night. It was *late* last night though so hopefully the thought process isn't gibberish:
-someone visits the site, anyone, and a new Order is created in the db and then loaded into the front end to be used in the redux store *and* the orderId is saved in local storage.
-If that person logs in later, their Order is updated with their userId when they check out (I don't think I updated the route to reflect this idea yet but it was laaaate)
-I'm not currently in favor of using the status flag as anything but a way to differentiate completed orders from active ones but could totally be convinced otherwise. I'm 100% ready to compromise lmao